tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PageFilterTest.java

```diff
@@ -0,0 +1,163 @@
+
+import org.apache.activemq.artemis.api.core.RoutingType;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.impl.AddressInfo;
+import org.apache.activemq.artemis.core.settings.impl.AddressSettings;
+import org.apache.activemq.artemis.tests.util.ActiveMQTestBase;
+import org.apache.activemq.artemis.tests.util.CFUtil;
+import org.apache.activemq.artemis.tests.util.PrintDataUtil;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PageFilterTest extends ActiveMQTestBase {
+
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+   public static final SimpleString ADDRESS = new SimpleString("SimpleAddress");
+
+   private ActiveMQServer server;
+
+   protected boolean isNetty() {
+      return true;
+   }
+
+   @Override
+   @Before
+   public void setUp() throws Exception {
+      super.setUp();
+      server = newActiveMQServer();
+
+      server.start();
+      waitForServerToStart(server);
+   }
+
+   private ActiveMQServer newActiveMQServer() throws Exception {
+      ActiveMQServer server = createServer(true, isNetty());
+
+      AddressSettings defaultSetting = new AddressSettings().setPageSizeBytes(10 * 1024).setMaxSizeMessages(10);
+
+      server.getAddressSettingsRepository().addMatch("#", defaultSetting);
+
+      return server;
+   }
+
+   @Test
+   public void testPageFilter() throws Exception {
+
+      ConnectionFactory connectionFactory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:61616");
+
+      int numberOfMessages = 3 * 24;
+      String filter = "partial=true";
+      String negfilter = "partial=false";
+
+      server.addAddressInfo(new AddressInfo(getName()).addRoutingType(RoutingType.MULTICAST));
+
+
+
+      try (Connection connection = connectionFactory.createConnection();
+           Connection otherConnection = connectionFactory.createConnection()) {
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         Topic topic = session.createTopic(getName());
+         session.createSharedDurableConsumer(topic, "wholeThing", negfilter);
+         MessageConsumer partialThing = session.createSharedDurableConsumer(topic, "partialThing", filter);
+
+         final String oneThousandCharacters;
+         {
+            StringBuilder builder = new StringBuilder(1000);
+            for (int i = 0; i < 1000; i++) {
+               builder.append('a');
+            }
+            oneThousandCharacters = builder.toString();
+         }
+
+         MessageProducer producer = session.createProducer(topic);
+
+
+         Session txsession = otherConnection.createSession(true, Session.SESSION_TRANSACTED);
+         MessageProducer txProducer = txsession.createProducer(topic);
+         for (int i = 0; i < 100; i++) {
```

Comment:
hard coding 100 here when numberOfMessages is defined to the rather random looking 3 * 24 = 72 at the start just seems confusing.